src/bincode/bincode.zig

```diff
@@ -345,14 +345,27 @@ pub fn readIntAsLength(comptime T: type, reader: anytype, params: bincode.Params
     return @intCast(len_u64);
 }
 
-pub fn readUtf8String(allocator: std.mem.Allocator, reader: anytype, _: bincode.Params) ![]const u8 {
-    const len = try reader.readInt(u64, .little);
-    const str = try allocator.alloc(u8, len);
-    errdefer allocator.free(str);
-    const bytes_read = try reader.readAll(str);
-    if (bytes_read != len) return error.NoBytesLeft;
-    if (!std.unicode.utf8ValidateSlice(str)) return error.InvalidUtf8;
-    return str;
+pub fn utf8StringCodec(
```

Comment:
Why the word "Codec" in this context? Think I'm missing something

Reply:
I am hoping to move us toward using the term "codec" rather than "FieldConfig", as it more accurately reflects what it really is: something that describes the encode and decode behavior of the type, be it for a field, or the type itself.

Reply:
Hm, wouldn't this imply that a codec should be both a serialiser+deserializer?